pkg/ebpf/c/common/filtering.h

```diff
@@ -323,13 +344,99 @@ statfunc u64 match_scope_filters(program_data_t *p)
     return res & policies_cfg->enabled_policies;
 }
 
+// Function to evaluate data filters based on the program data and index.
+// Returns policies bitmap.
+//
+// Parameters:
+// - program_data_t *p: Pointer to the program data structure.
+// - u8 index: Index of the string data to be used as filter.
+statfunc u64 match_data_filters(program_data_t *p, u8 index)
+{
+    policies_config_t *policies_cfg = &p->event->policies_config;
+    u64 res = ~0ULL;
+
+    u16 version = p->event->context.policies_version;
+    void *filter_map = NULL;
+
+    if (!(policies_cfg->data_filter_exact_enabled || policies_cfg->data_filter_prefix_enabled ||
+          policies_cfg->data_filter_suffix_enabled)) {
+        return policies_cfg->enabled_policies;
+    }
+
+    // key_prefix is allocated to store string data used for
+    // exact and prefix matching
+    data_filter_lpm_key_t *key_prefix = get_data_filter_buf(DATA_FILTER_BUF1_IDX);
+    if (key_prefix == NULL)
+        return 0;
+
+    // get event ID
+    key_prefix->event_id = p->event->context.eventid;
+
+    // get string based on index
+    __builtin_memset(key_prefix->str, 0, sizeof(key_prefix->str));
```

Comment:
Do we really need this step if a step after we copy into `key_prefix->str`?
Since we are giving the string length to the lpm trie lookup - I don't think it is actually required to zero the unused bytes